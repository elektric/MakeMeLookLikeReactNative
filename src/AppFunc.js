import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  Switch,
} from 'react-native';

export default class AppFuncs extends Component {
  constructor(props){
    super(props);
    this.state = {
            switchOn: true,
          };
  }
  render() {
    return (
      <View style={styles.splitViewOuter}>
        <View style={styles.splitViewInnerLeft}>
            <Text style={styles.appfuncStyle}>
              {this.props.details.Name}(
              {this.props.details.AuthLevel})
            </Text>
            <Text style={styles.appfuncDescriptionStyle}>
              {this.props.details.Description}
            </Text>
          </View>
          <View style={styles.splitViewInnerRight}>
          <Switch style={styles.splitViewInnerRight} onValueChange={(value) => this.setState({switchOn: value})}
            value={this.state.switchOn} />
          </View>
        </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {

    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    //backgroundColor: 'powderblue',
    borderBottomColor: '#bbb',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  slider: {

  },
  splitViewOuter: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  splitViewInnerLeft: {
    flex: 1,
    //justifyContent: 'center',
  },
  splitViewInnerRight: {
    //flex: 1,
    //justifyContent: 'center',
  },
  userName: {
   fontSize: 20,
   color: '#51b',
   marginBottom: 8,
   textAlign: 'center',
   //backgroundColor: 'skyblue',

 },
 userID: {
   color: '#51b',
   fontSize: 30,
   textAlign: 'center',
   //backgroundColor: 'steelblue',
 },
 appfuncStyle: {
   color: '#511',
   fontSize: 16,
   textAlign: 'center',
   marginBottom: 1,
   //backgroundColor: 'steelblue',
 },
 appfuncDescriptionStyle: {
   color: '#951',
   fontSize: 12,
   textAlign: 'center',
   marginBottom: 10,
   //backgroundColor: 'steelblue',
 },
});
