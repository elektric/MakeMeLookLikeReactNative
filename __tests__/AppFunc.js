import 'react-native';
import React from 'react';
import AppFunc from '../src/AppFunc';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('AppFunc renders', () => {
  let af = {'Name':'AppFunc2','AuthLevel':'9204','Description':'Application Function 2 Description Text Sometimes this text might be really really really super duper long so we should definately test it out when it is long.'};
  const tree = renderer.create(
    <AppFunc details={af} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
