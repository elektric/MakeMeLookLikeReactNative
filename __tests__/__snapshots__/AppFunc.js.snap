// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFunc renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#511",
          "fontSize": 16,
          "marginBottom": 1,
          "textAlign": "center",
        }
      }
    >
      AppFunc2
      (
      9204
      )
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#951",
          "fontSize": 12,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Application Function 2 Description Text Sometimes this text might be really really really super duper long so we should definately test it out when it is long.
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <RCTSwitch
      disabled={false}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {},
        ]
      }
      value={true}
    />
  </View>
</View>
`;
